include/logger.h

```diff
@@ -8,14 +8,22 @@
 
 // * Log record : each line of the log
 // * log format : tuple like <Date, Name, Message>
-// * log entity : Date, Name, Message
+// * log property : Date, Name, Message
 #define LOG(severity)                                                          \
   logger::Logger::getInstance().log<logger::severity>(                         \
-      {__func__, __FILE__, __LINE__})
+      {__FILE__, __func__, __LINE__})
+
+
+std::string hex(intptr_t v) {
+  std::stringstream ss;
+  ss << std::hex << "0x" << v;
+  return ss.str();
+}
 
 namespace logger {
 
 // TODO: how to specify THRESHOLD in terms of these? make two header files? NO!
+// TODO: JEB suggested we should put these in the log config
```

Comment:
Sounds good to me... maybe we can try to move them back and resolve the circular dependency in another way.